prediction_market_agent/agents/autogen_general_agent/social_agent.py

```diff
@@ -130,11 +140,23 @@ def build_social_media_text(model: str, bets: list[Bet]) -> str:
         trigger=writer,
     )
 
-    # ToDO - fetch agent reasoning from DB and construct better tweets
-    #  See https://github.com/gnosis/prediction-market-agent/issues/150
+    learnings = ""
+    if long_term_memory:
+        memories = long_term_memory.search(from_=memories_since)
+        simple_memories = [
+            SimpleMemoryThinkThoroughly.from_long_term_memory(ltm) for ltm in memories
+        ]
+        # We want memories only from the bets to add relevant learnings
+        questions_from_bets = set([b.market_question for b in bets])
+        filtered_memories = [
+            m for m in simple_memories if m.metadata.question in questions_from_bets
+        ]
```

Comment:
We match the question from the bet with the question from the learning - note that each learning has a `scenario` (i.e. the subscenario generated by the `think-thoroughly` agent) and a `question`, which is the parent scenario.
By matching the question we avoid having summaries that don't correspond to the bets we placed.